cli/001-cli-improve-piping-support.md

```diff
@@ -0,0 +1,33 @@
+- Start Date: 2021-06-11
+- RFC PR:
+- Issue:
+
+### Support better piping of output 
+Problem: currently piping the output into other tools is not well supported. For example utilities such as `grep`, `wc`, `jq` may work with very limited set of commands / in limited functionality. 
+
+Structure outputs so it would be easy to pipe it into common utilities. 
+
+To achieve that:
+
+ - `wc`, `more` support: integrate output with `less` cmd by default if it's available in the OS. If `less` is not found, create and use builtin functiniality to output all available rows immediately (similar to how `cat` behaves).
```

Comment:
> TBH I'm not sure we want to automatically pipe to pager

do you mean by default to use a simple builtin pager implementation? Then if people want to use another pager like `less` they would use pipe syntax (ie. ` | less`)? 

In case of piping ` | less` is this actually possible to have lazy pagination over data? (and not first prefetch all data)

Reply:
my understanding was that it's not possible to lazy load with piping syntax (at least quick google search didn't show anything), and then going over other modern CLIs i've found that they mostly all use internal piping into these tools 

ex. 
aws - https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html
bat - https://github.com/sharkdp/bat/blob/91a347bf6d549ba543f1b4bfdc2a05f269c975c8/src/less.rs#L7